Content.Server/DeltaV/CartridgeLoader/Cartridges/MailMetricsCartridgeSystem.cs

```diff
@@ -0,0 +1,98 @@
+using Content.Server.DeltaV.Cargo.Components;
+using Content.Server.DeltaV.Cargo.Systems;
+using Content.Server.Station.Systems;
+using Content.Server.CartridgeLoader;
+using Content.Shared.CartridgeLoader;
+using Content.Shared.CartridgeLoader.Cartridges;
+using Content.Server.Mail.Components;
+
+namespace Content.Server.DeltaV.CartridgeLoader.Cartridges;
+
+public sealed class MailMetricsCartridgeSystem : EntitySystem
+{
+    [Dependency] private readonly CartridgeLoaderSystem _cartridgeLoader = default!;
+    [Dependency] private readonly LogisticStatsSystem _logisticsStats = default!;
+    [Dependency] private readonly StationSystem _station = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<MailMetricsCartridgeComponent, CartridgeUiReadyEvent>(OnUiReady);
+        SubscribeLocalEvent<LogisticStatsUpdatedEvent>(OnLogisticsStatsUpdated);
+        SubscribeLocalEvent<MailComponent, ComponentInit>(OnComponentInit);
+    }
+
+    private void OnUiReady(Entity<MailMetricsCartridgeComponent> ent, ref CartridgeUiReadyEvent args)
+    {
+        UpdateUI(ent, args.Loader);
+    }
+
+    private void OnLogisticsStatsUpdated(LogisticStatsUpdatedEvent args)
+    {
+        UpdateAllCartridges(args.Station);
+    }
+
+    private void OnComponentInit(EntityUid uid, MailComponent mail, ComponentInit args)
+    {
+        var stationUid = _station.GetOwningStation(uid);
+        if (stationUid != null)
+            UpdateAllCartridges((EntityUid) stationUid);
+    }
+
+    private void UpdateAllCartridges(EntityUid station)
+    {
+        var query = EntityQueryEnumerator<MailMetricsCartridgeComponent, CartridgeComponent>();
+        while (query.MoveNext(out var uid, out var comp, out var cartridge))
+        {
+            if (cartridge.LoaderUid is not { } loader || comp.Station != station)
+                continue;
+            UpdateUI((uid, comp), loader);
+        }
+    }
+
+    private void UpdateUI(Entity<MailMetricsCartridgeComponent> ent, EntityUid loader)
+    {
+        if (_station.GetOwningStation(loader) is { } station)
+            ent.Comp.Station = station;
+
+        if (!TryComp<StationLogisticStatsComponent>(ent.Comp.Station, out var logiStats))
+            return;
+
+        // Get station's logistic stats
+        var mailEarnings = logiStats.MailEarnings;
+        var damagedMailLosses = logiStats.DamagedMailLosses;
+        var expiredMailLosses = logiStats.ExpiredMailLosses;
+        var tamperedMailLosses = logiStats.TamperedMailLosses;
+        var openedMailCount = logiStats.OpenedMailCount;
+        var damagedMailCount = logiStats.DamagedMailCount;
+        var expiredMailCount = logiStats.ExpiredMailCount;
+        var tamperedMailCount = logiStats.TamperedMailCount;
+        var unopenedMailCount = GetUnopenedMailCount();
```

Comment:
put all these stats in a struct that the ui state and component both use instead of doing giga copy paste